### PR TITLE
Move require statements outside of factory function

### PR DESCRIPTION
**Pull request by captbaritone**
This is a prerequisite for resolving #1.

Review @msrose

### CODE REVIEW

**Comment by msrose** [comment]
(Will also have to update README before this merges.)

File: index.js [hunk 1]
```diff
@@ -81,15 +81,13 @@ module.exports = ({ types: t }) => {
...
+            newNodes.push(createRequireExpression(el, paramName));
           });
         }
 
         if(factory) {
           const factoryArity = factory.params.length;
-          let replacementFuncExpr = t.functionExpression(null, [], t.blockStatement(
-            requireExpressions.concat(factory.body.body)
-          ));
+          let replacementFuncExpr = t.functionExpression(null, [], t.blockStatement(factory.body.body));
```
Comment (captbaritone) [inline index.js]: We need to remove the arguments, so we still need the create a new function expression. However, upon closer inspection, we don't need to create a new block statement. Can reuse the factory function's existing one.

**Comment by captbaritone** [comment]
I've made my updates as individual commits to aid review. If you would prefer, I'm happy to squash them down to a single commit before we merge.

**Comment by msrose** [comment]
Great, I will likely just use the github squash+merge option when merging so no need to worry about doing it on your branch. 

Also, I've just thought of the following scenario with these changes:
```js
require(['one'], function(one) {
  require(['two'], function(one) {
    // ...
  });
  one.doStuff();
});
```
Right now it gets transformed to
```js
(function() {
  var one = require('one');
  (function() {
    var one = require('two');
    // ...
  })();
  one.doStuff();
})();
```
But with inserting the requires before it becomes:
```js
var one = require('one');
(function() {
  var one = require('two');
  (function() {
    // ...
  })();
  one.doStuff();
})();
```
Which ends up changing the value of `one` for the `one.doStuff()` line.
While it's pretty unlikely that this will happen since it needs nonsensical variable naming, it is a gotcha to look out for with nested `require`s... Is it worth doing something like wrapping transformed `require`s in an extra IIFE? Or inserting the dependencies into the factory body for `require`, but outside for `define`? 

I'm fine to merge this PR as-is, document the gotcha with the other ones, and leave it for a later date.

**Comment by msrose** [comment]
We've instead gone with #8 to close #1.

**Pull request closed without merging**